### PR TITLE
Fixes cached router view holder's alpha issue

### PR DESCRIPTION
**Pull request by suresh-gangumalla**
with default transition, alpha is set to zero for previous route view holder element. if preivous route is configured to keep alive then that view is cached with holder alpha set to zero

Now, if previous route has a custom transition and if it does not contain alpha related prop transition in 'before' or 'in' then holder's alpha will not get updated to 1 from zero set by previous route out transition

### CODE REVIEW

**Review by jfboeve — COMMENTED** [review]
I recommend changing the variable name `isAlphaPropExists` -> `hasAlphaProp`

**Comment by michielvandergeest** [comment]
> I recommend changing the variable name `isAlphaPropExists` -> `hasAlphaProp`

yes I agree. @suresh-gangumalla can you make the change accordingly?

File: src/router/router.js [hunk 1]
```diff
@@ -168,6 +168,27 @@ export const navigate = async function () {
...
+              }
+            }
+          } else {
+            if (route.transition.before.prop === 'alpha') {
+              hasAlphaProp = true
+            }
+          }
+        }
+        // set holder alpha when alpha prop is not exists in route transition
+        if (!hasAlphaProp) {
```
Comment (michielvandergeest) [inline src/router/router.js:189]: please do a `false` check instead of a `falsey` check here

File: src/router/router.js [hunk 2]
```diff
@@ -168,6 +168,27 @@ export const navigate = async function () {
...
+          if (Array.isArray(route.transition.before)) {
+            for (let i = 0; i < route.transition.before.length; i++) {
+              if (route.transition.before[i].prop === 'alpha') {
+                hasAlphaProp = true
+                break
+              }
+            }
+          } else {
+            if (route.transition.before.prop === 'alpha') {
+              hasAlphaProp = true
```
Comment (michielvandergeest) [inline src/router/router.js:184]: this could be combined in an `else if`?

**Review by michielvandergeest — CHANGES_REQUESTED** [review]

**Review by michielvandergeest — APPROVED** [review]
LGTM

**Pull request merged**